components/input/gpio/gpio.go

```diff
@@ -267,7 +267,6 @@ func (c *Controller) newButton(ctx context.Context, brd board.Board, interrupt b
 
 	c.activeBackgroundWorkers.Add(1)
 	utils.ManagedGo(func() {
-		defer interrupt.RemoveCallback(tickChan)
```

Comment:
Sorry I can't comment on the lines I have questions about. Without this line, the callbacks added from `StreamTicks` on line 263 will be removed when `ctx` is canceled. This function is only called from `NewGPIOController`, and the context passed in here is a descendant of the context passed to that function. Doesn't that context get canceled immediately when the constructor returns? I suspect line 101 should actually be `context.WithCancel(context.Background())` instead.